1449 improve examples (#1661)
open-contracting/standard: docs/primer/how.md

In addition to the default format of JSON, you can convert and publish your OCDS data as Excel and CSV (comma-separated values) files.

````{admonition} Publication formats: JSON and Excel/CSV
:class: hint

While the OCDS schema is described using JSON Schema, OCDS data can be converted from its JSON format to tabular formats such as CSV files or spreadsheets. JSON is favored by developers because it uses human-readable text to exchange complex information, such as nested objects. It can contain large volumes of information and is particularly good at handling one-to-many relationships (such as multiple bids per tender notice).

> Tabular formats, such as CSV (or comma separated values) are commonly used in spreadsheets applications and other analysis tools. Many people are comfortable working with spreadsheets using tools like Excel. While JSON is the default format, a good publication will publish tabular formats as well both so that more users’ needs can be satisfied.

Reviewer: Correcting a couple of typos I noticed.

```suggestion
Tabular formats, such as CSV (or comma separated values) are commonly used in spreadsheet applications and other analysis tools. Many people are comfortable working with spreadsheets using tools like Excel. While JSON is the default format, a good publication will publish tabular formats both so that more users’ needs can be satisfied.
```

**Example**

> Ebonyi State in Nigeria publish OCDS data in JSON and tabular format. The JSON data is flattened into a series of CSV files which are combined into a single Excel spreadsheet. 

Reviewer: Copy-edit first sentence. Remove second sentence because the purpose of this example is to illustrate what the different publication formats look like rather than to explain Ebonyi State's implementation.
 
```suggestion
Nigeria's Ebonyi State publishes OCDS data in JSON and tabular formats.
```

> The following example does not contain all existing fields, and not all of the included fields are required. It has been edited for length and consistency, and is not identical to the data available on the Ebonyi State e-Procurement platform.

Reviewer: I'm not sure that we need this disclaimer. James's comment in https://github.com/open-contracting/standard/issues/1449#issuecomment-1729874835 was that we aren't making a promise that the data is identical. I think that's implicit in framing something as an example.

If we do need a disclaimer for this example then presumably we need one for all the other examples based on real data so I think it would make more sense to include a generic statement at the bottom of the docs landing page along the lines of:

> The example data featured in the OCDS documentation are based on real publications and edited where needed for brevity, clarity and correctness.

Author: I agree that it should be implicit in the framing that something is an example and therefore not necessarily identical. I think it's worth including a disclaimer though to avoid (a probably unlikely tbf circumstance) of someone from e.g. Ebonyi State seeing this and feeling we've somehow misrepresented them by changing their data but not explicitly saying that's what we've done.

Author: re-reading and yes I agree that a generic disclaimer on the docs landing page would work, @jpmckinney what do you think?

Reviewer 2: Yes, we can put a simple statement somewhere about the examples being edited for clarity.

@yolile Do we have preferences re: which publishers to take examples from? I'm not sure about the longevity of some publishers (and some publishers have very small or limited datasets). e.g. any examples in the Primer should be best-in-class in case anyone is inspired to look them up elsewhere.

Beginning to wonder if this issue is more trouble than it's worth.

Reviewer 3: As far as I know, all the OCDS publishers have more or less some data issues, either quality, modeling, structure, longevity, access, etc. That is why I wasn't sure about referencing real examples in the first place, so I don't have a strong preference for any of them

Reviewer 2: Okay, maybe, @duncandewhurst, you can re-read #1449 and see if any of the existing examples mentioned are deficient, and if so, propose some improvement?

Reviewer: @jpmckinney I've done that in https://github.com/open-contracting/standard/issues/1449#issuecomment-1849353843

Reviewer: Resolving as this PR no longer adds examples from real publishers.

**JSON**

```{jsoninclude} ../examples/primer/primer.json
:jsonpointer:
:expand: releases, items
```

**CSV**

***releases***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/releases.csv
```

***parties***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/parties.csv
```

***tender_items***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/tender_items.csv
```

***documents***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/documents.csv
```

***awards***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/awards.csv
```

***award_suppliers***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/award_suppliers.csv
```

***award_items***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/award_items.csv
```

***contracts***

```{csv-table-no-translate}
:header-rows: 1
:file: ../examples/primer/contracts.csv
```

***contract_implementation***

```{csv-table-no-translate}
:header-rows: 1

> :file: ../examples/primer/contract_implementation.csv

Reviewer: I think that including all of the fields and CSV tables will overwhelm readers. The purpose of the example is to illustrate what the different formats look like so I would omit the release package and everything but the required release- and tender-level fields, plus a small subset of fields belonging to the tender object, e.g. `title`, `status`, `procurementMethod`, `mainProcurementCategory`, `tenderPeriod`.

Alternatively, you could borrow the approach used in [What does OFDS data look like?](https://open-fibre-data-standard.readthedocs.io/en/latest/primer/openfibredatastandard.html#what-does-ofds-data-look-like) and include only the releases table in the example.

Author: I think including more than one table is important to make it clear that a single json file is likely to be equivalent to more than one csv file/spreadsheet sheet. But yes to cutting down on the amount of data.

Author: I've removed the release package and everything high level except `tender` and `parties` which isn't required. I've left `parties` to ensure there's 2 files.

In addition to the default format of JSON, you can convert and publish your OCDS data as Excel and CSV (comma-separated values) files.

````{admonition} Publication formats: JSON and Excel/CSV
:class: hint

While the OCDS schema is described using JSON Schema, OCDS data can be converted from its JSON format to tabular formats such as CSV files or spreadsheets. JSON is favored by developers because it uses human-readable text to exchange complex information, such as nested objects. It can contain large volumes of information and is particularly good at handling one-to-many relationships (such as multiple bids per tender notice).

> Tabular formats, such as CSV (or comma separated values) are commonly used in spreadsheet applications and other analysis tools. Many people are comfortable working with spreadsheets using tools like Excel. While JSON is the default format, a good publication will publish tabular formats both so that more users’ needs can be satisfied.

Reviewer 2: ```suggestion
Tabular formats, such as CSV (or comma separated values) are commonly used in spreadsheet applications and other analysis tools. Many people are comfortable working with spreadsheets using tools like Excel. While JSON is the default format, a good publication will publish tabular formats as well, so that more users’ needs can be satisfied.
```
